### PR TITLE
Set `ignore_masking` to True by default

### PR DESCRIPTION
**Pull request by sararb**
Fixes https://github.com/NVIDIA-Merlin/NVTabular/issues/1688 

- This is a quick fix to set `ignore_masking` to True in the model's forward method.
- The motivation behind this fix is that during prediction/inference, we always want to use the whole input sequence without any masking.

### CODE REVIEW

File: transformers4rec/torch/model/prediction_task.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def calculate_metrics(  # type: ignore
 
         outputs = {}
         if forward:
-            predictions = self(predictions)
+            predictions = self(predictions, ignore_masking=False)
```
Comment (rnyak) [inline transformers4rec/torch/model/prediction_task.py:277]: why False?

File: transformers4rec/torch/features/sequence.py [hunk 2]
```diff
@@ -247,7 +247,7 @@ def item_embedding_table(self) -> Optional[torch.nn.Module]:
 
         return None
 
-    def forward(self, inputs, training=True, ignore_masking=False, **kwargs):
+    def forward(self, inputs, training=True, ignore_masking=True, **kwargs):
```
[the remark this answers is not in the document]
Comment (sararb) [reply transformers4rec/torch/features/sequence.py:250]: [Here](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/73b1e31634443596fdd52a3bccdfbc0a2f9b50a1/transformers4rec/torch/trainer.py#L397) The training and evaluation scripts (in the Trainer class) set the parameter ` ignore_masking` to `False` to ensure masking is leveraged when calling `forward()`. 

The example notebooks are using the high-level Trainer class (that sets the flag to the correct value based on whether it's training/evaluation or prediction model) so no need to change the example notebooks.

File: transformers4rec/torch/model/prediction_task.py (hunk 1 above)
Comment (sararb) [reply transformers4rec/torch/model/prediction_task.py:277]: To compute metrics and losses we need to compute the masked targets.

File: tests/torch/test_losses.py [hunk 3]
```diff
@@ -19,12 +19,12 @@ def test_item_prediction_with_label_smoothing_ce_loss(
...
     )
 
-    body_outputs = body(torch_yoochoose_like)
+    body_outputs = body(torch_yoochoose_like, ignore_masking=False)
 
     trg_flat = input_module.masking.masked_targets.flatten()
     non_pad_mask = trg_flat != input_module.masking.padding_idx
     labels_all = pytorch.masked_select(trg_flat, non_pad_mask)
-    predictions = head(body_outputs)
+    predictions = head(body_outputs, ignore_masking=False)
```
[the remark this answers is not in the document]
Comment (sararb) [reply tests/torch/test_losses.py:27]: It is set to `False` to compute the masked targets, needed for computing the loss. So in that case, `ignore_masking` should be equal to `False` for any masking method.

**Comment by karlhigley** [comment]
Looks good from a serving standpoint 👍🏻

**Review by rnyak — APPROVED** [review]
